thirdeye/thirdeye-frontend/app/pods/custom/dimensions-table/dimension/component.js

```diff
@@ -15,44 +16,101 @@
  * }
  */
 import Component from "@ember/component";
-import { getWithDefault } from '@ember/object';
-import { get, set, setProperties } from '@ember/object';
+import { computed, get, set, setProperties, getWithDefault } from '@ember/object';
+import { isPresent } from '@ember/utils';
 import { inject as service } from '@ember/service';
 import { once } from '@ember/runloop'
 
-const BLOCK_CLASS = 'rootcause-dimensions-table__dimension-cell';
+const CUSTOM_CLASS = 'rootcause-dimensions-table__dimension-cell';
 
 export default Component.extend({
 
-  dimensionLabel: null,
+  /**
+   * The column's cell content - dimension value
+   * @type {String}
+   */
+  dimensionValue: null,
+
+  /**
+   * The dimension group's list of excluded dimensions, typically present in a dimension of value '(ALL)-'
+   * @type {String}
+   */
+  excludedDimensions: null,
+
+  /**
+   * Visibility of content, based on whether its a dupe of the cell above it
+   * @type {Boolean}
+   */
+  isValueShown: true,
+
+  /**
+   * Determines whether the current cell will have the "blank" class applied
+   * @type {Boolean}
+   */
   isCellHidden: false,
-  isRowSelected: false,
-  onSelection: null,
-  inputId: null,
-  classNames: [BLOCK_CLASS],
-  classNameBindings: [`isCellHidden:${BLOCK_CLASS}--blank`],
 
-  didReceiveAttrs() {
+  /**
+   * Component classes
+   * @type {Array}
+   */
+  classNames: [CUSTOM_CLASS, `${CUSTOM_CLASS}--dynamic`],
+
+
+  init() {
     this._super(...arguments);
-    this._formatDimensionLabel();
+
+    once(() => {
```

Comment:
is it running more than once without this?

Reply:
It was running more than once, but not for this reason. I've corrected it by adding refresh logic in `init`.